### PR TITLE
feat(router-plugin): add `RouterDataResolved` action

### PR DESCRIPTION
**Pull request by markwhitfeld**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Issue Number: #946 #922 #1049 
Progression of PR #952 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by markwhitfeld** [comment]
@arturovt turns out that the test failure was caused by a race condition encountered if jasmine ran the tests in a specific order. The race condition was caused by the case if testA was run before testB and both tests used the same route. As a result, testB had code that was waiting for the route to change after navigating to the url, but this was never triggered because the document was already located at that url!

File: packages/router-plugin/src/router.state.ts [hunk 1]
```diff
@@ -118,8 +116,13 @@ export class RouterState {
     });
   }
 
-  private navigationEnd(): void {
-    this.routerStateSnapshot = this._serializer.serialize(this.lastResolvedStateSnapshot);
+  private guardsCheckEnd(routerState: RouterStateSnapshot): void {
+    if (this.routerStateSnapshot) {
+      // routerState = { ...this.routerStateSnapshot, ...routerState };
```
Comment (splincode) [inline packages/router-plugin/src/router.state.ts]: Is it really worth leaving comments in the code?

Comment (markwhitfeld) [reply packages/router-plugin/src/router.state.ts]: Ah, that was Artur's fix, but it turned out that it was not needed. I'll remove it.

**Comment by szalonna** [comment]
Shouldn't the [`RouterAction`](https://github.com/ngxs/store/blob/master/packages/router-plugin/src/router.actions.ts#L79) type contain the `RouterDataResolved` action?

**Pull request merged**